scripts/src/public/services/offers.ts

```diff
@@ -71,15 +72,19 @@ async function filterOffers(userId: string, offers: db.Offer[], logger: LoggerIn
 export async function getOffers(userId: string, appId: string, filters: ModelFilters<db.Offer>, logger: LoggerInstance): Promise<OfferList> {
 	let offers = [] as Offer[];
 
+	const query = Application.createQueryBuilder("app")
+		.where("app.id = :appId", { appId })
+		.leftJoinAndSelect("app.offers", "offer");
+
 	if (!filters.type || filters.type === "earn") {
 		offers = offers.concat(
 			await filterOffers(
 				userId,
-				await db.Offer.createQueryBuilder()
-					.where("type = 'earn'")
-					.orderBy("amount", "DESC")
-					.addOrderBy("id", "ASC")
-					.getMany(),
+				(await query
+					.andWhere("offer.type = :type", { type: "earn" })
```

Comment:
why not:
```.andWhere("offer.type = 'earn'")```

Reply:
1. It's safer as you don't have to remember the quotes.
2. It's good practice to use parameters for both performance and security reasons